### PR TITLE
Refactor init/server

### PR DESCRIPTION
**Pull request by therealplato**
omg this is brutal

i have about a dozen commits where i updated all packages in the same commit

i'm going through these commits and stripping out everything that's not in the server package and will do it again for other packages

in the future i'm going to try to modify only one package per commit

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -334,48 +307,47 @@ func TestPostCommandDeployBad(t *testing.T) {
...
 	req, w := testutils.DeleteRequest(
 		t,
 		fmt.Sprintf("/instances/%s/%s", testInstance1.PlaybookID, testInstance1.ID),
 	)
 
-	req = auth(req)
-	makeRequest(req, w)
+	req = auth(testCfg, req)
+	e := New(testCfg, ets).Handler()
+	// _, _, e := helperSetupServer(testCfg)
```
Comment (therealplato) [inline server/server_test.go]: :boom:

File: server/server.go [hunk 2]
```diff
@@ -51,18 +52,19 @@ func CustomError(message string) ErrorResponse {
 
 // New instantiates a new Server and binds its handlers. The Server will look
 // for playbooks and instances in store `s`
-func New(s store.Store) *Server {
+func New(cfg cfg.Type, s store.Store) *Server {
 	srvr := &Server{
+		Cfg:        cfg,
 		store:      s,
-		slackToken: env.SlackToken,
+		slackToken: cfg.SlackToken,
```
Comment (bobbytables) [inline server/server.go]: TODO: Refactor out.

File: server/server.go [hunk 3]
```diff
@@ -101,16 +103,18 @@ func (s *Server) Run(addr ...string) error {
...
-	if len(a) == 0 || a != env.AuthBearerToken {
-		glog.Infof("Auth failure for %s\nExpected: %s Actual: %s\n", c.Request.URL.Path, env.AuthBearerToken, a)
-		c.String(http.StatusUnauthorized, "Wrong or Missing Authorization")
-		c.AbortWithStatus(http.StatusUnauthorized)
-		return
+func (s *Server) genAuthMiddleware() func(c *gin.Context) {
+	return func(c *gin.Context) {
+		a := c.Request.Header.Get("Authorization")
+		a = strings.TrimPrefix(a, "Bearer ")
+		if len(a) == 0 || a != s.Cfg.AuthBearerToken {
```
Comment (bobbytables) [inline server/server.go]: Let's allow requests when the config does _not_ have a BearerToken

```golang
if a != s.Cfg.AuthBearerToken {
}
```

**Comment by bobbytables** [comment]
Minus one comment, 👍

**Comment by poopoothegorilla** [comment]
+1

**Pull request merged**